mesonbuild/backend/ninjabackend.py

```diff
@@ -2169,6 +2169,17 @@ def generate_rust_target(self, target: build.BuildTarget) -> None:
             self.generate_shsym(target)
         self.create_target_source_introspection(target, rustc, args, [main_rust_file], [])
 
+        if target.doctests:
+            from ..compilers.rust import RustCompiler
+            assert isinstance(rustc, RustCompiler)
```

Comment:
I think it would be fine to just do a `T.cast` here, and not need to import RustCompiler. We have a pretty good idea that this is a RustCompiler. If you really want to be sure `assert rustc.language == 'rust'` should be sufficient, right?